tests/test_utils.py

```diff
@@ -303,3 +306,437 @@ def check_multistream_classic(fnames):
+
+        # remove spaces from column names in DataFrame
+        vic_42 = vic_42.rename(columns=lambda x: x.replace(' ', ''))
+
+        # load VIC 5.0 data
+
+        vic_50_file = 'outfile_%s_%s.txt' % (lat, lng)
+
+        vic_50 = pd.read_csv(os.path.join(vic_50_dir, vic_50_file),
+                                skiprows=3,
+                                sep='\t')
+
+        # remove spaces from column names
+        vic_50 = vic_50.rename(columns=lambda x: x.replace(' ', ''))
+
+        # load VIC 5.0.x data
+
+        vic_50x_file = 'outfile_%s_%s.txt' % (lat, lng)
+
+        vic_50x = pd.read_csv(os.path.join(result_dir, vic_50x_file),
+                                skiprows=3,
+                                sep='\t')
+
+        # remove spaces from column names
+        vic_50x = vic_50x.rename(columns=lambda x: x.replace(' ', ''))
+
+        # variables to plot
+        plot_variables = ['OUT_SWE', 'OUT_ALBEDO', 'OUT_SALBEDO', 'OUT_SNOW_DEPTH',
+                        'OUT_SNOW_CANOPY', 'OUT_SNOW_PACK_TEMP', 'OUT_SNOW_MELT',
+                        'OUT_R_NET', 'OUT_LATENT', 'OUT_SENSIBLE']
+
+        plot_units = ['mm', 'fraction', 'fraction', 'mm', '%', 'degrees C',
+                        'mm', 'W/$m^2$', 'W/$m^2$', 'W/$m^2$']
+
+        len_dates = len(snotel_swe['DATES'])
+
+        # plotting preferences
+        lw = 4.0
+        # loop over variables to plot
+
+        for i, plot_variable in enumerate(plot_variables):
+
+            if 'water_year' in plots_to_make:
+
+                plt.figure(figsize=(10,10))
+
+                if plot_variable == "OUT_SWE":
+
+                    # plot SnoTel SWE observations
+                    plt.plot(snotel_swe['DATES'], snotel_swe[plot_variable],
+                            'k', label='Snotel', linewidth=lw)
+
+                # plot VIC 4.2 simulations
+                plt.plot(snotel_swe['DATES'], vic_42[plot_variable][:len_dates],
+                        'b', label='VIC 4.2', linewidth=lw)
+
+                # plot VIC 5.0 simulations
+                plt.plot(snotel_swe['DATES'], vic_50[plot_variable][:len_dates],
+                        'r', label='VIC 5.0', linewidth=lw)
+
+                # plot VIC 5.0.x simulations
+                plt.plot(snotel_swe['DATES'], vic_50x[plot_variable][:len_dates],
+                        'y', label='VIC 5.0.x', linewidth=lw)
+
+                plt.title(plot_variable)
+                plt.legend(loc='upper left')
+                plt.ylabel(plot_units[i])
+
+                # save figure
+                os.makedirs(os.path.join(plot_dir, plot_variable), exist_ok=True)
+                plotname = '%s_%s.png' % (lat, lng)
+                savepath = os.path.join(plot_dir, plot_variable, plotname)
+                plt.savefig(savepath, bbox_inches='tight')
+
+                plt.clf()
+                plt.close()
+
+
+def plot_fluxnet_comparison(driver, testname, result_dir, plot_dir, vic_42_dir, vic_50_dir,
```

Comment:
this function is really long and is somewhat hard to follow.  I think you want to break it out into a few pieces.

1.  get obs locations
2.  read obs data
3.  loop over model datasets and read them
4.  append obs data so that you have an iterable of like shaped series (`datasets.keys() == ['Observations', 'VIC.4.2.d', 'VIC.5.0.0.rc1', 'VIC.5.0.0.rc2', 'VIC.5.0.0', 'VIC.5.1.0']`)
5.  do analysis and make plots
  - annual cycle
  - diurnal cycle
  - other?